src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1818,13 +1819,47 @@ export default abstract class ComparisonStore
         },
     });
 
+    readonly survivalSequencedMonths = remoteData<
+        { [uniquePatientKey: string]: number } | undefined
+    >({
+        await: () => [this.studies],
+        invoke: async () => {
+            const studyIds = this.studies.result!.map(s => s.studyId);
+            if (studyIds.length === 1 && studyIds[0] === 'nsclc_genie_bpc') {
+                const allTimelineData = _.flatten(
+                    await Promise.all(
+                        studyIds.map(studyId =>
+                            internalClient.getAllClinicalEventsInStudyUsingGET({
+                                studyId,
+                            })
+                        )
+                    )
+                );
+                const sequencingData: ClinicalEvent[] = allTimelineData.filter(
+                    d => d.eventType === 'Sequencing'
+                );
+                return sequencingData.reduce(
```

Comment:
it's not great that you are fetching the clinican events and then throwing away all the data except that which you need.  this makes it impossible to resuse that clincail events for other purposes.   instead, have one promise that just load the event and then do the processing of the data in your business logic.